+nosnoc/+dcs/Gcs.m

```diff
@@ -0,0 +1,67 @@
+classdef Gcs < nosnoc.dcs.Base
+    properties
+        lambda
+
+        f_x
+
+        dims
+
+        f_fun
+        nabla_c_fun
+        c_fun
+        g_comp_path_fun
+    end
+
+    methods
+        function obj = Gcs(model)
+            obj.model = model;
+            obj.dims = model.dims;
+        end
+        
+        function generate_variables(obj, opts)
+            import casadi.*
+            dims = obj.dims;
+            % dimensions
+            dims.n_lambda = dims.n_c;
+            obj.lambda = define_casadi_symbolic(opts.casadi_symbolic_mode,'lambda',dims.n_lambda);
+
+            obj.dims = dims;
+        end
+
+        function generate_equations(obj, opts)
+            import casadi.*
+            model = obj.model;
+            dims = obj.dims;
+
+            nabla_c = model.c.jacobian(model.x)';
+            
+            obj.f_x = model.f_x + model.E*nabla_c*obj.lambda;
+
+            obj.f_x_fun = Function('f_x', {model.x, model.z, obj.lambda, model.u, model.v_global, model.p}, {obj.f_x, model.f_q});
+            obj.f_fun = Function('f', {model.x, model.z, model.u, model.v_global, model.p}, {model.f_x});
+            obj.f_q_fun = Function('f_q', {model.x, model.z, obj.lambda, model.u, model.v_global, model.p}, {model.f_q});
+            obj.g_z_fun = Function('g_z', {model.x, model.z, model.u, model.v_global, model.p}, {model.g_z});
+            obj.g_alg_fun = Function('g_alg', {model.x, model.z, obj.lambda, model.u, model.v_global, model.p}, {[]});
+            obj.c_fun = Function('c_fun', {model.x, model.z, model.v_global, model.p}, {model.c});
+            obj.nabla_c_fun = Function('c_fun', {model.x, model.z, model.v_global, model.p}, {nabla_c});
+            obj.g_path_fun = Function('g_path', {model.x, model.z, model.u, model.v_global, model.p}, {model.g_path}); % TODO(@anton) do dependence checking for spliting the path constriants
```

Comment:
regarding the todo, split in what? (e.g. if they depend only on control, but you have path on all rk stages, to avoid duplicating constraints, or is it something else)

Reply:
Yes precisely this. We can do this via depends on but it is some effort to add this improvement hence the TODO.

Reply:
I am not sure should  we make all these TODOs Issues. I am afraid that we forget them if they live in stable code that we don't look at often.

Reply:
Added issue #112